healthz/checks/checks.go

```diff
@@ -25,29 +25,38 @@ import (
 
 	"github.com/gravitational/satellite/agent/health"
 	pb "github.com/gravitational/satellite/agent/proto/agentpb"
-	"github.com/gravitational/satellite/healthz/config"
 	"github.com/gravitational/satellite/monitoring"
 )
 
-// RunAll runs all checks successively and reports general cluster status
-func RunAll(cfg config.Config) (*pb.Probe, error) {
-	var probes health.Probes
-	var checkers health.Checkers
+// Runner stores configures checkers and provides interface to configure
+// and run them
+type Runner struct {
+	health.Checkers
+}
 
-	checkers.AddChecker(monitoring.KubeAPIServerHealth(cfg.KubeAddr))
-	checkers.AddChecker(monitoring.NodesStatusHealth(cfg.KubeAddr))
-	etcdChecker, err := monitoring.EtcdHealth(&cfg.ETCDConfig)
+// NewRunner creates Runner with checks configured using provided options
+func NewRunner(kubeAddr string, etcdConfig monitoring.ETCDConfig) (*Runner, error) {
+	runner := &Runner{}
+	runner.AddChecker(monitoring.KubeAPIServerHealth(kubeAddr))
+	runner.AddChecker(monitoring.NodesStatusHealth(kubeAddr))
+	etcdChecker, err := monitoring.EtcdHealth(&etcdConfig)
```

Comment:
Move this to the beginning of the method - no point creating the runner if this fails